From 72ae7ce646792b07061cbbb31ed47f55a1d2422c Mon Sep 17 00:00:00 2001
Message-Id: <72ae7ce646792b07061cbbb31ed47f55a1d2422c.1663075350.git.yann.morin@orange.com>
From: "Yann E. MORIN" <yann.morin@orange.com>
Date: Tue, 13 Sep 2022 15:10:48 +0200
Subject: [PATCH] gpsctl: fix yodaification

Commit d5a672abd98a (gpsctl.c: Style tweaks.  Yoda style, braces, //
comments.) claimed "No functional changes", but mis-treated a comparison
when it was converted to yoda-style (simplified diff):

    -    if (write(gpsdata->gps_fd, buf, strlen(buf)) <= 0) {
    +    if (0 <= write(gpsdata->gps_fd, buf, strlen(buf))) {

As one may notice, the comparison operator was not reversed while the
operands were.

This got silently fixed in commit 3219ab9f6a89 (gpsctl.c: First cut at
--ship sending commands to running gpsd.) but this is completely
unrelated.

Fix that comparison.

Signed-off-by: Yann E. MORIN <yann.morin@orange.com>
Upstream status: not applicable as already fixed.
---
 gpsctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gpsctl.c b/gpsctl.c
index cb342af87..ef59a6622 100644
--- a/gpsctl.c
+++ b/gpsctl.c
@@ -96,7 +96,7 @@ static bool gps_query(struct gps_data_t *gpsdata,
     if ('\n' != buf[strnlen(buf, sizeof(buf) - 1) - 1]) {
         (void)strlcat(buf, "\n", sizeof(buf));
     }
-    if (0 <= write(gpsdata->gps_fd, buf, strnlen(buf, sizeof(buf)))) {
+    if (0 >= write(gpsdata->gps_fd, buf, strnlen(buf, sizeof(buf)))) {
         GPSD_LOG(LOG_ERROR, &context.errout, "gps_query(), write failed\n");
         return false;
     }
-- 
2.25.1

